ext/drx/drx.h

```diff
@@ -435,6 +435,19 @@ DR_EXPORT
 size_t
 drx_buf_get_buffer_size(void *drcontext, drx_buf_t *buf);
 
+
+DR_EXPORT
+/**
+ * Pads a basic block with a label for routines which rely on inserting instrumentation
+ * after every instruction. Note that users of this routine must act on the previous
+ * instruction before the is-app check afterwards, because the label is not marked as an
+ * app instruction.
+ *
+ * \returns whether terminator was introduced.
```

Comment:
The client cannot insert instru after a call or syscall, but the client could do cleanup, and it would simplify this drx routine -- but it sounds like it is messier this way so I agree, let's go back to what you originally had.